tests/integration/test_main.py

```diff
@@ -24,19 +26,27 @@ async def wait_to_receive(dispatcher, ct, timeout=5.0, interval=0.05):
 async def test_run_lambda_function(apg_dispatcher, pg_message):
     assert apg_dispatcher.pool.finished_count == 0
 
-    clearing_task = asyncio.create_task(apg_dispatcher.pool.events.work_cleared.wait())
+    clearing_task = asyncio.create_task(apg_dispatcher.pool.events.work_cleared.wait(), name='test_lambda_clear_wait')
     await pg_message('lambda: "This worked!"')
     await asyncio.wait_for(clearing_task, timeout=3)
 
     assert apg_dispatcher.pool.finished_count == 1
 
 
 @pytest.mark.asyncio
-async def test_run_decorated_function(apg_dispatcher, conn_config):
-    assert apg_dispatcher.pool.finished_count == 0
+async def test_run_decorated_function(apg_dispatcher, test_settings):
+    clearing_task = asyncio.create_task(apg_dispatcher.pool.events.work_cleared.wait())
+    test_methods.print_hello.apply_async(settings=test_settings)
```

Comment:
Just for reference: https://github.com/ansible/dispatcher/issues/83, not a blocker for merging. 

I believe you mentioned at some point that you don't want to run checkers on tests. I would agree that in some cases can be annoying or not useful, but in that cases we can just use exceptions/ignores
I think treating the test code with the same rules has some benefits: on one hand the test would allow us to consume the library as a client and detect issues that we could not detect in other way as well as enhance the development or debugging of tests.

Reply:
I agree we should run checkers on tests, I just don't believe they should be the same checkers as on the code. Tests need to pass objects of the wrong type... because they're testing what it does in that case.

https://github.com/ansible/dispatcher/issues/83 seems like a little bit of a different subject to me, and it is a real problem. Firstly, it seems we need to do the obvious of completing type hints for the methods here. After that, I'm unclear about what the best practice would be. There's a lot you need to do to carry over methods that you wrap, but `@task` doesn't technically wrap.